Modules/readline.c

```diff
@@ -1347,15 +1347,17 @@ call_readline(FILE *sys_stdin, FILE *sys_stdout, const char *prompt)
     if (should_auto_add_history && n > 0) {
         const char *line;
         int length = _py_get_history_length();
-        if (length > 0)
+        if (length > 0) {
+            HIST_ENTRY *hist_ent;
 #ifdef __APPLE__
             if (using_libedit_emulation) {
                 /* handle older 0-based or newer 1-based indexing */
-                line = (const char *)history_get(length + libedit_history_start - 1)->line;
+                hist_ent = history_get(length + libedit_history_start - 1);
             } else
 #endif /* __APPLE__ */
-            line = (const char *)history_get(length)->line;
-        else
+                hist_ent = history_get(length);
+            line = (const char *)hist_ent ? hist_ent->line : "";
```

Comment:
The cast is not needed. A compiler shouldn't emit a warning when assigning `char *` to `const char *`.

And more, I have doubts about the priority of the cast and the trinary operator. Does the above code is equivalent to `(const char *)(hist_ent ? hist_ent->line : "")` or to `((const char *)hist_ent) ? hist_ent->line : ""`?